File: apollo-runtime/src/main/java/com/apollographql/apollo/subscription/SubscriptionConnectionParamsProvider.java
package com.apollographql.apollo.subscription;

import org.jetbrains.annotations.NotNull;

/**
 * Provides instance of {@link SubscriptionConnectionParams} to be sent to the subscription server after connection is established.
 */
public interface SubscriptionConnectionParamsProvider {
  @NotNull
  SubscriptionConnectionParams provide();

  class Const implements SubscriptionConnectionParamsProvider {
    private final SubscriptionConnectionParams params;

    public Const(@NotNull SubscriptionConnectionParams params) {
      this.params = params;
    }

    @NotNull
    @Override
    public SubscriptionConnectionParams provide() {
      return params;
    }
  }
}

File: deprecated-apollo-runtime/src/main/java/com/apollographql/apollo3/subscription/SubscriptionConnectionParamsProvider.kt
package com.apollographql.apollo3.subscription

/**
 * Provides instance of [SubscriptionConnectionParams] to be sent to the subscription server after connection is established.
 */
interface SubscriptionConnectionParamsProvider {
  fun provide(): SubscriptionConnectionParams
  class Const(private val params: SubscriptionConnectionParams) : SubscriptionConnectionParamsProvider {
    override fun provide(): SubscriptionConnectionParams {
      return params
    }
  }
}